python_ta/checkers/one_iteration_checker.py

```diff
@@ -25,20 +26,59 @@ class OneIterationChecker(BaseChecker):
     # pass in message symbol as a parameter of check_messages
     @check_messages('one-iteration')
     def visit_for(self, node):
-        if self._check_one_iteration(node):
+        if hasattr(node, 'cfg_block') and self._check_one_iteration(node):
```

Comment:
This check is added because `While` and `For` nodes defined in the class scope don't have a control flow graph. To avoid this altogether (and for all other checks using the CFG), I think it would be good to have a control flow graph for class definitions. If you (@david-yz-liu) think so too, then I'll work on implementing it.

Reply:
@kavins14 Sorry I missed this earlier. The `ClassDef` node shouldn't have its own CFG, but the statements in it should be part of the CFG of the "main" module. (This is actually how the control flow works in Python; statements in the class body are indeed evaluated when the ClassDef is reached.)

So in fact, nothing special should be done; in the `ClassDef` visitor, you can just remove the check that looks for `FunctionDef` nodes, and call `accept` with all the nodes in the body.

Reply:
(And therefore revert the change on this line)

Reply:
https://github.com/pyta-uoft/pyta/pull/600